uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/API/Query/LeaveRequestSelect.php

```diff
@@ -217,4 +221,39 @@ private function addGroupBy(CRM_Utils_SQL_Select $query) {
     $query->groupBy(['a.id']);
   }
 
+  /**
+   * Returns if the balance_change field should be included on the returned results
+   *
+   * @return bool
+   */
+  private function shouldReturnBalanceChange() {
+    return $this->shouldReturnField('balance_change');
+  }
+
+  /**
+   * Returns if the dates field should be included on the returned results
+   *
+   * @return bool
+   */
+  private function shouldReturnDates() {
+    return $this->shouldReturnField('dates');
+  }
+
+  /**
+   * Returns, based on the "return" param, if the given field should be returned
+   * on the results.
+   *
+   * If "return" is empty, it means all the fields should be returned. Otherwise,
+   * a field will be returned only if "return" is not empty and it includes the
+   * field.
+   *
+   * @param string $field
+   *
+   * @return bool
+   */
+  private function shouldReturnField($field) {
+    return empty($this->params['return']) ||
+           (is_array($this->params['return']) && in_array($field, $this->params['return']));
```

Comment:
The problem is that `$this->params['return']` might not defined, so I have to check if it's not empty first, otherwise I will get a E_NOTICE. If I do as you say, the method will have to look like this:

```php
  private function shouldReturnField($field) {
    $returnParamHasField = !empty($this->params['return']) && 
                           is_array($this->params['return']) && 
                           in_array($field, $this->params['return']);

    return empty($this->params['return']) || $returnParamHasField;
  }
```

Which is more complext then the original.